library/std/src/diagnostics.qs

```diff
@@ -149,5 +149,73 @@ namespace Microsoft.Quantum.Diagnostics {
         areEqual
     }
 
-    export DumpMachine, DumpRegister, CheckZero, CheckAllZero, Fact, CheckOperationsAreEqual;
+    /// # Summary
+    /// Starts counting the number of times the given operation is called. The counter is reset to zero.
+    ///
+    /// # Description
+    /// This operation allows you to count the number of times a given operation is called. As part of
+    /// starting teh counting, the counter is reset to zero, which may override existing counts for the same operation.
+    /// Counting is based on the specific specialization of the operation invoked, so `X` and `Adjoint X` are counted separately.
+    /// Likewise `Controlled X`, `CNOT`, and `CX` are independent operations that are counted separately.
```

Comment:
What about Adjoint Controlled and Controlled Adjoint, are they the same or different?

Reply:
Those are the same as each other but different from the others. So you must count body, adjoint, controlled, and controlled adjoint (which is the same as adjoint controlled) separately. But this comment made me realize that I should not treat `Controlled Controlled` and different than `Controlled`, so I will make that change and add a unit test.